community/kernel/src/main/java/org/neo4j/kernel/impl/core/NodeProxy.java

```diff
@@ -377,6 +382,85 @@ public Object getProperty( String key, Object defaultValue )
     }
 
     @Override
+    public Map<String, Object> getProperties( String... keys )
```

Comment:
I think we should throw if `keys == null` or if `keys` contains `null` values.
We can return `emptyMap()` when `keys.length == 0`.